### PR TITLE
Add docs about configuring Json formatters

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/11838

### CODE REVIEW

File: aspnetcore/web-api/advanced/formatting.md [hunk 1]
```diff
@@ -95,30 +100,51 @@ services.AddMvc(options =>
...
+
+```csharp
+services.AddMvc()
+    .AddNewtonsoftJson();
+```
+
+Some features may not work very well with System.Text.Json based formatters and require a reference to the Newtonsoft.Json based formatters for the ASP.NET Core 3.0
+release. Here are some reasons to continue using the Newtonsoft.Json based formatters in your ASP.NET Core 3.0 applications:
+
+* Your application uses Newtonsoft.Json attributes (e.g. `JsonProperty`, `JsonIgnore` etc), customizes the serialization settings , or relies on features that
```
Comment (pranavkm) [inline aspnetcore/web-api/advanced/formatting.md]: cc @rynowak

File: aspnetcore/web-api/advanced/formatting.md [hunk 2]
```diff
@@ -95,30 +100,51 @@ services.AddMvc(options =>
...
+* Your application extensively uses <xref:Microsoft.AspNetCore.Mvc.JsonResult>. `JsonResult` is a Newtonsoft.Json specific `IActionResult` and will not work without a reference to it.
+* Your application generates [OpenAPI](<xref:tutorials/web-api-help-pages-using-swagger>) documentation.
+
+::: moniker-end
+
+### Adding XML Format Support
+
+To add support for XML formatting, install the `Microsoft.AspNetCore.Mvc.Formatters.Xml` NuGet package.
+
+Add the XmlSerializerFormatters to MVC's configuration in *Startup.cs*:
```
Comment (scottaddie) [inline aspnetcore/web-api/advanced/formatting.md]:
```suggestion
Add the XML serializer formatters to MVC's configuration in `Startup.ConfigureServices`:
```

**Comment by scottaddie** [comment]
@pranavkm Ping. Anything I can do to help with this PR?

File: aspnetcore/web-api/advanced/formatting.md [hunk 3]
```diff
@@ -95,30 +102,50 @@ services.AddMvc(options =>
...
-});
+services.AddMvc()
+    .AddNewtonsoftJson();
+```
+
+Some features may not work well with `System.Text.Json`-based formatters and require a reference to the `Newtonsoft.Json`-based formatters for the ASP.NET Core 3.0
+release. Continue using the `Newtonsoft.Json`-based formatters if your ASP.NET Core 3.0 or later app:
+
+* Uses `Newtonsoft.Json` attributes (for example, `[JsonProperty]` or `[JsonIgnore]`), customizes the serialization settings, or relies on features that `Newtonsoft.Json` provides.
+* Configures <xref:Microsoft.AspNetCore.Mvc.JsonResult.SerializerSettings>. Prior to 3.0, `JsonResult.SerializerSettings` accepts an instance of `JsonSerializerSettings` that is specific to Newtonsoft.Json.
```
Comment (pranavkm) [inline aspnetcore/web-api/advanced/formatting.md]: @scottaddie tweaked this line a bit.

File: aspnetcore/web-api/advanced/formatting.md [hunk 4]
```diff
@@ -95,30 +95,50 @@ services.AddMvc(options =>
...
+
+### Add XML format support
+
+To add XML formatting support, install the [Microsoft.AspNetCore.Mvc.Formatters.Xml](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Formatters.Xml/) NuGet package.
+
+Add the XML serializer formatters to MVC's configuration in `Startup.ConfigureServices`:
+
+[!code-csharp[](./formatting/sample/Startup.cs?name=snippet1&highlight=2)]
+
+These two approaches will serialize results using `System.Xml.Serialization.XmlSerializer`. If you prefer, you can use the `System.Runtime.Serialization.DataContractSerializer` by adding its associated formatter in `Startup.ConfigureServices`:
```
Comment (scottaddie) [inline aspnetcore/web-api/advanced/formatting.md]: It isn't clear to my what the 2 approaches are. Can this be clarified?

File: aspnetcore/web-api/advanced/formatting.md [hunk 5]
```diff
@@ -95,30 +95,50 @@ services.AddMvc(options =>
...
 
-To add support for XML formatting, install the `Microsoft.AspNetCore.Mvc.Formatters.Xml` NuGet package.
+### Configure System.Text.Json-based formatters
 
-Add the XmlSerializerFormatters to MVC's configuration in *Startup.cs*:
-
-[!code-csharp[](./formatting/sample/Startup.cs?name=snippet1&highlight=2)]
-
-Alternately, you can add just the output formatter:
+Features for the `System.Text.Json`-based formatters can be configured using <xref:Microsoft.AspNetCore.Mvc.MvcOptions.SerializerOptions>.
```
Comment (scottaddie) [inline aspnetcore/web-api/advanced/formatting.md]: Is `SerializerOptions` a new API in 3.0? I ask because this xref link doesn't work.

Comment (pranavkm) [reply aspnetcore/web-api/advanced/formatting.md]: Yes, this would be new in 3.0.

File: aspnetcore/web-api/advanced/formatting.md (hunk 4 above)
Comment (pranavkm) [reply aspnetcore/web-api/advanced/formatting.md]: I copied the text from before, but yeah, one of those isn't really correct.

File: aspnetcore/web-api/advanced/formatting.md [hunk 6]
```diff
@@ -95,30 +95,50 @@ services.AddMvc(options =>
...
+
+* Uses `Newtonsoft.Json` attributes (for example, `[JsonProperty]` or `[JsonIgnore]`), customizes the serialization settings, or relies on features that `Newtonsoft.Json` provides.
+* Configures <xref:Microsoft.AspNetCore.Mvc.JsonResult.SerializerSettings>. Prior to 3.0, `JsonResult.SerializerSettings` accepts an instance of `JsonSerializerSettings` that is specific to `Newtonsoft.Json`.
+* Generates [OpenAPI](<xref:tutorials/web-api-help-pages-using-swagger>) documentation.
+
+::: moniker-end
+
+### Add XML format support
+
+To add XML formatting support, install the [Microsoft.AspNetCore.Mvc.Formatters.Xml](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Formatters.Xml/) NuGet package.
```
Comment (pranavkm) [inline aspnetcore/web-api/advanced/formatting.md]: How does this look?

File: aspnetcore/web-api/advanced/formatting.md (hunk 5 above)
Comment (scottaddie) [reply aspnetcore/web-api/advanced/formatting.md]: I've removed the link until the 3.0 APIs have been published.

**Comment by scottaddie** [comment]
Thank you for all your help, @pranavkm!

**Pull request merged**